connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -337,6 +330,71 @@ object KustoWriter {
     CsvRowResult(fields.toArray, size + fields.size)
   }
 
+  private def convertOneField(row: InternalRow, schemaFields: Array[StructField], fieldIndexInRow: Int, timeZone: String) = {
+    val dateFormat = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ss.SSSXXX", TimeZone.getTimeZone(timeZone))
+    val dataType = schemaFields(fieldIndexInRow).dataType
+    dataType match {
+      case DateType => DateTimeUtils.toJavaDate(row.getInt(fieldIndexInRow)).toString
+      case TimestampType => dateFormat.format(DateTimeUtils.toJavaTimestamp(row.getLong(fieldIndexInRow)))
+      case StringType => row.getString(fieldIndexInRow)
+      case BooleanType => row.getBoolean(fieldIndexInRow).toString
+      case _ : StructType => convertStructToCsv(row.getStruct(fieldIndexInRow, dataType.asInstanceOf[StructType].length), dataType.asInstanceOf[StructType], timeZone)
+      case _ : ArrayType => ConvertArrayToCsv(row.getArray(fieldIndexInRow), dataType, timeZone)
+      case _ => row.get(fieldIndexInRow, dataType).toString
+    }
+  }
+
+  private def convertStructToCsv(row: InternalRow, schema: StructType, timeZone: String): String = {
+    val dateFormat = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ss.SSSXXX", TimeZone.getTimeZone(timeZone))
+    val fields = schema.fields
+    val result:scala.collection.mutable.Map[String,String] = scala.collection.mutable.Map()
+
+    for (x <- 0 to fields.length - 1) {
+      val dataType = fields(x).dataType
+      val value = if (row.get(x, fields(x).dataType) == null) null else dataType match {
```

Comment:
why do you need to call row.get here - when you call the explicit "get" typed function later